### DISCUSSION TITLE
New JS rendering tool for Crawly

### DISCUSSION THREAD

**Discussion by oltarasenko** [elixir-crawly/crawly#273] [Ideas]
It has become quite tricky to use Splash now. I was not able to compile it from sources due to some dependencies differences between Python and its deps versions :(. And the already published Docker images are not compatible with M1 Mac computers.

I was trying to address this problem with the help of NodeJS, where I am not an expert. However, the tool I have made seems to work; see more: https://github.com/elixir-crawly/crawly/pull/272

It would be great to hear your feedback!

**Comment by cmnstmntmn** [comment]
@oltarasenko I might be offtopic, but somehow it related to node; Recently I was looking over https://github.com/mozilla/readability .. that is implemented in FF (other browsers have something simillar). What is actually doing is to process the page and render a minimal version of it.

Would this help crawly to speed up things and users to create simpler element queries.

## Normal 

<img width="1392" alt="normal" src="https://github.com/elixir-crawly/crawly/assets/2312755/21de19ea-e953-46c5-9669-73a6ecb3089a">

## Reader view

<img width="1392" alt="reader_view" src="https://github.com/elixir-crawly/crawly/assets/2312755/f5d30c6e-420b-45ff-bded-c32e1a2f2593">
